source/services/encryption/EncryptionService.ts

```diff
@@ -80,3 +76,31 @@ export async function decryptFormAnswers(user: User, forms: Forms) {
     return forms;
   }
 }
+
+export async function setupSymmetricKey(user: UserInterface, forms: FormsInterface) {
+  const otherUserPersonalNumber = Object.keys(forms.encryption.publicKey.publicKeys).filter(
+    (key) => key !== user.personalNumber
+  )[0];
+
+  const otherUserPublicKey = getPublicKeyInForm(otherUserPersonalNumber, forms);
+  let ownPublicKey = getPublicKeyInForm(user.personalNumber, forms);
+
+  if (!ownPublicKey) {
+    const privateKey = await createAndStorePrivateKey(user, forms);
+    ownPublicKey = getPseudoKey(
+      forms.encryption.publicKey.G,
+      privateKey,
+      forms.encryption.publicKey.P
+    );
+
+    forms.encryption.publicKey.publicKeys[user.personalNumber] = ownPublicKey;
```

Comment:
This modifies `forms` in-place I believe, which can have unwanted side-effects. `forms` should be deep-copied and the copy should be modified instead (and then returned from this function).

Reply:
You are absolutely correct! Sloppy code.